app/src/main/java/com/nutomic/syncthingandroid/activities/MainActivity.java

```diff
@@ -475,28 +477,36 @@ private void setOptimalDrawerWidth(View drawerContainer) {
     /**
      * Displays dialog asking user to accept/deny usage reporting.
      */
-    private void showUsageReportingDialog() {
+    private void showUsageReportingDialog(RestApi restApi) {
         final DialogInterface.OnClickListener listener = (dialog, which) -> {
-            Options options = getApi().getOptions();
-            switch (which) {
-                case DialogInterface.BUTTON_POSITIVE:
-                    getApi().getSystemInfo(systemInfo -> {
-                        options.urAccepted = systemInfo.urVersionMax;
-                    });
-                    break;
-                case DialogInterface.BUTTON_NEGATIVE:
-                    options.urAccepted = Options.USAGE_REPORTING_DENIED;
-                    break;
-                case DialogInterface.BUTTON_NEUTRAL:
-                    Uri uri = Uri.parse("https://data.syncthing.net");
-                    startActivity(new Intent(Intent.ACTION_VIEW, uri));
-                    break;
+            if (restApi == null) {
```

Comment:
No point for this check as this does not get called if restApi is null in the outer stack.

Reply:
Careful, this is callback code, so restApi may be null when the user "sits there some minutes and run conditions shut down syncthingservice + restApi in the meantime".

Reply:
Well no, as rest API is passed in as an argument, and before being passed as an argument, there is an if that checks that it's not null.

Reply:
Ok, but as this is a delayed callback I suggest getting rid of https://github.com/syncthing/syncthing-android/pull/1155/commits/408e6dafc3f7f6c50ce7f993b5d0601d0cf50c53 instead.

Reply:
Sorry but how is this async?

![image](https://user-images.githubusercontent.com/1144861/41512815-815a9a68-7288-11e8-8cef-34e55307748b.png)

Reply:
see line 481 in your screenshot. The onClickListener will fire some time later when the user clicked on the answer button in the dialog. So line 482 and so on are not guaranteed to have a valid restApi not being null available.

Reply:
Well, you are guaranteed to have a non-null RestAPI, as that was passed to the function, and that's the rest api used in scope of the lambda.

Reply:
Ok, picked up your feedback in commit https://github.com/syncthing/syncthing-android/pull/1155/commits/c5ed92218995dfa8b4c1b4f3ba4001e9fc47025e